src/community/community.mongo.service.ts

```diff
@@ -0,0 +1,76 @@
+import { HttpException, HttpStatus, Injectable, Logger } from '@nestjs/common';
+import { Model } from 'mongoose';
+import { InjectModel } from '@nestjs/mongoose';
+import { Community, CommunityDocument } from './community.schema';
+import { CommunityCreateDto, CommunityUpdateDto } from './community.dtos';
+
+@Injectable()
+export class CommunityMongoService {
+    private readonly logger = new Logger(CommunityMongoService.name);
+
+    constructor(
+        @InjectModel(Community.name) private communityModel: Model<CommunityDocument>,
+    ) {
+        // do nothing
+    }
+
+    async createCommunity(communityCreateDto: CommunityCreateDto): Promise<Community> {
+        this.logger.log('Creating new community in db');
+
+        try {
+
+            return await this.communityModel.create(communityCreateDto);
+
+        } catch (e) {
+
+            this.logger.error('Failed to create community in db', e);
+
+            throw new HttpException('Failed to create community in db', HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    async fetchAllCommunities(): Promise<Community[]> {
+        try {
+            return await this.communityModel.find().exec();
```

Comment:
maybe should include some privacy option? Some communities may not want to be shown 🤔

Reply:
I think that is something we should evaluate for all endpoints at some point. Right now they are all behind the same api key but we can have different policies for each. Some endpoints are actually more for utility and testing and could either be locked for clients or removed entirely.